apps/distgo/cmd/publish/github_publish.go

```diff
@@ -102,13 +87,31 @@ func (g *GitHubConnectionInfo) Publish(buildSpec params.ProductBuildSpec, paths
 }
 
 func (g *GitHubConnectionInfo) uploadFileAtPath(client *github.Client, release *github.RepositoryRelease, filePath string, stdout io.Writer) (string, error) {
+	// create asset upload URI using upload template provided in release creation
+	// response
+	//
+	// API: https://developer.github.com/v3/repos/releases/#response
+	//
+	// Examples:
+	//   - https://uploads.github.com/repos/octocat/Hello-World/releases/1/assets{?name,label}
+	//   - https://github.enterprise.host/api/uploads/repos/octocat/Hello-World/releases/1/assets{?name,label}
+	t, err := uritemplates.Parse(release.GetUploadURL())
```

Comment:
Let's factor this out into a separate functions and then add tests with the example inputs above